numpy/ma/testutils.py

```diff
@@ -91,6 +91,21 @@ def _assert_equal_on_sequences(actual, desired, err_msg=''):
     return
 
 
+def _assert_equal_both_non_masked(actual, desired, err_msg=''):
+    """
+    Asserts the equality of two non masked arrays
+
+    """
+    if (isinstance(desired, (list, tuple)) and
+            isinstance(actual, (list, tuple))):
+        return _assert_equal_on_sequences(actual, desired, err_msg)
+    elif (isinstance(actual, np.matrix) or
+            isinstance(desired, np.matrix)):
+        return assert_array_equal(actual, desired, err_msg)
+    else:
+        return utils.assert_equal(actual, desired, err_msg)
```

Comment:
I think this is the only function you need. `utils.assert_equal` checks sequences/tuples and handles the matrix case as matrix is a subclass or ndarray.